scripts/check_whitespace.sh

```diff
@@ -26,6 +26,7 @@ options=(
 )
 
 git grep "${options[@]}" -- \
+    ':(exclude)cmake/external/snappy.patch' \
```

Comment:
The file format of "patch" files contain trailing whitespace as part of their syntax. It is not only valid, but required. Therefore, patch files need to be excluded from the trailing whitespace checks.

If you were to run `scripts/check_whitespace.sh Firestore` without this change, it would fail with the following error:
```
cmake/external/snappy.patch:6: 
ERROR: Trailing whitespace found in the files above. Please fix.
```